R/jointRPCA.R

```diff
@@ -0,0 +1,616 @@
+#' Joint RPCA front-end and orchestration
+#'
+#' User-facing wrappers and internal orchestration of Joint Robust Principal Component Analysis (Joint-RPCA)
+#' for compositional multi-omic data using the OptSpace algorithm.
+#'
+#' This file contains:
+#' \itemize{
+#'   \item \code{jointRPCAuniversal()} — universal front-end detecting input type
+#'         (\code{matrix}, \code{list}, \code{SummarizedExperiment}, or \code{MultiAssayExperiment}).
+#'   \item \code{jointRPCA()} — main Joint-RPCA function handling preprocessing,
+#'         shared-sample alignment, factorization, and output assembly.
+#'   \item \code{.joint_optspace_helper()} — internal coordination of multi-view OptSpace factorization.
+#'   \item \code{.transform()} and \code{.transform_helper()} — projection of new data into existing ordination space.
+#'   \item \code{.rpca_table_processing()} — compositional preprocessing (filtering, zero-removal, etc.).
+#'   \item \code{.mask_value_only()} — safe numeric matrix masking for missing values.
+#'   \item Other small utility functions used across the Joint-RPCA workflow.
+#' }
+#'
+#' These functions depend on the OptSpace backend defined in \code{optspace.R}.
+#' Only \code{jointRPCA()} and \code{jointRPCAuniversal()} are exported;
+#' all others are internal helpers.
+#'
+#' @keywords internal
+#' @noRd
+NULL
+
+#' Universal Joint RPCA Wrapper
+#'
+#' @param data Input object: MultiAssayExperiment, TreeSummarizedExperiment, SummarizedExperiment, list of matrices, or single matrix.
+#' @param assays Character vector of assay names to extract (only used for MultiAssayExperiment). If NULL, use all assays.
+#' @param ... Additional arguments passed to jointRPCA()
+#'
+#' @return Output from jointRPCA()
+#' @export
+jointRPCAuniversal <- function(data, assays = NULL, ...) {
```

Comment:
I suggest to rename "data" argument to just "x". This is in line with the usual R/Bioc conventions.